beamline.rst

```diff
@@ -1,212 +1,379 @@
+        # That means that for collection you need beamstop IN, and frontlight OUT
+        #
+        # NB Do we need an (oxymoronic) third state, like UNUSABLE?
+
+        INACTIVE = 0
+        OUT = 0
+        OPEN = 0
+
+        ACTIVE = 1
+        IN = 1
+        CLOSED = 1
+
+
+Specific Actuators:
+-------------------
+
+**Type float:**
+
+    *Centring motors*
+
+        Omega, Kappa, Phi, AlignmentX,  AlignmentY,  AlignmentZ,
+        CentringX, CentringY, CentringZ,  Focus
+
+        # NB, the capitalized case is to distinguish them from the current
+        #     lower-case names ('Phi' matches 'kappa_phi', not 'phi')
+        #
+        # NB There are too many motor names here, and maybe we should shorten
+        #    the list. It should be OK, though if the same motor is pointed to
+        #    by two different names. The problems is how the names match.
+        #    If you do not have 'CentringZ', which motor should you use during
+        #    centring?
+        #    Is 'Focus' always the same as 'AlignmentX'?
+        #    Does 'CentringZ' always match the same alignment motor
+        #    or does it depend on beamline geometry?
+
+    *Detector motors*
+
+        detector_distance, two_theta, detector_horizontal, detector_vertical
+
+        # detector_distance may be the only common one,
+        # but the others are clearly defined.
+        # NB horizontal and vertical are kept separate, in case some beamlines
+        #    have one without the other.
+
+    *Beamline parameters*
+
+        resolution, energy, wavelength, transmission
+
+        # NB energy/wavelength and resolution/detector_distance are both needed.
+
+    *Light intensity*
+
+        frontlight_intensity, backlight_intensity
+
+        # NB are these continuous-value floats, or rather values_list or on/off?
+
+
+**Type Tuple[float, float]:**
+
+    *Standard parameters*
+
+        beam_size, beam_position
+
+        # These should be generally supported
+
+    *Expert parameters*
+
+        aperture, slits, beam_definer
+
+        # These are included to standardize the names, but the specific
+        # beam-defining motors are likely to vary between beamlines.
+
+
+**Type str:**
+
+    *Enumerated strings*
+
+        zoom, phase, centring_method, beam_shape
+
+        # These must all have a values_list or enum
+        # NB we should standardise the vocabulary. An enum??
+
+**Type TwoState**
+
+    fast_shutter, safety_shutter, beamstop, capillary, frontlight, backlight
+
+    # These are all two-state actuators. Their value will be of type
+    # TwoStateValue. Their state is set to ActuatorState, though one could
+    # limit it to a subset: NOTINITIALIZED, UNUSABLE, READY, MOVING, FROZEN
+    #
+    # NB it is open whether multistate objects should be handled similarly
+    #    or should be done as string enums?
+    #
+    # NB If e.g. a beamstop has multiple positions, how should we treat it?
+
+**Immovable actuators**
```

Comment:
So they re not actuators ...

Reply:
Yes. I know it is a bit of a stretch. But 1) It would be nice to use the same code to get e.g. the wavelength whether the beamline is tunable or not. 2) From a UI point of view e.g. photon_flux is a continuously updated value that is shown, that sends value_changed signals, that can be down (UNUSABLE), or working. It just cannot be modified from the UI.